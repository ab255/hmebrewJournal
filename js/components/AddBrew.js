import React, { Component } from 'react';
import {
  View,
  TextInput,
  Text,
  StyleSheet,
  TouchableHighlight,
  DatePickerIOS,
} from 'react-native';

export default class AddBrew extends Component {
  constructor(){
    super();
    this.state = {
      beerName: '',
      beerType: '',
      brewDate: new Date(),
      brewNotes: '',
    }
  }

  onDateChange = (date) => {
    this.setState({brewDate: date})
  }

  render() {
    return (
      <View>
        <Text>Beer Name:</Text>
        <TextInput
          style={styles.brewInput}
          onChangeText={(beerName) => this.setState({beerName})}
          value={this.state.beerName}
          placeholder='Beer Name'
        />
        <Text>Beer Style:</Text>
        <TextInput
          style={styles.brewInput}
          onChangeText={(beerType) => this.setState({beerType})}
          value={this.state.beerType}
          placeholder='Beer Style'
        />
        <Text>Brew Date:</Text>
          <DatePickerIOS
            date={this.state.brewDate}
            mode="date"
            onDateChange={this.onDateChange}
          />
        <Text>Brew Notes:</Text>
        <TextInput
          multiline={true}
a
          style={styles.brewNotesInput}
          onChangeText={(brewNotes) => this.setState({brewNotes})}
          value={this.state.brewNotes}
          placeholder='Take notes on your entire beer process'
        />
        <TouchableHighlight style={styles.submitButton}>
          <Text style={styles.buttonText}>Submit</Text>
        </TouchableHighlight>
      </View>
    );
  }
}


const styles = StyleSheet.create({
  brewInput: {
    height: 30,
    borderBottomWidth: 0.5,
    borderBottomColor: '#630A10',
  },
  brewNotesInput: {
    borderBottomColor: '#000',
    borderBottomWidth: 1,
    height: 100,
    fontSize: 16,
  },
  submitButton: {
    backgroundColor: '#630A10',
    width: 100,
  },
  buttonText: {
    color: '#FFF',
  },
})
