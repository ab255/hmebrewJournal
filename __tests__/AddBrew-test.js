import 'react-native';
import React from 'react';
import AddBrew from '../js/components/AddBrew.js';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <AddBrew />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
