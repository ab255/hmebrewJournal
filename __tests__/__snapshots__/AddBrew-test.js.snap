exports[`test renders correctly 1`] = `
<ScrollView
  contentInset={
    Object {
      "bottom": 49,
    }
  }
  enableAutoAutomaticScroll={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  onScroll={[Function]}
  scrollEventThrottle={8}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#911F27",
      "flex": 1,
      "padding": 10,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#F7D098",
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }>
    Beer Name:
  </Text>
  <TextInput
    autoCapitalize="words"
    onChangeText={[Function]}
    placeholder="Beer Name"
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 5,
        "color": "#630A10",
        "height": 30,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    value="" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#F7D098",
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }>
    Beer Style:
  </Text>
  <TextInput
    autoCapitalize="words"
    onChangeText={[Function]}
    placeholder="Beer Style"
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 5,
        "color": "#630A10",
        "height": 30,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    value="" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#F7D098",
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }>
    Ingredients:
  </Text>
  <TextInput
    multiline={true}
    onChangeText={[Function]}
    placeholder="Add ingredients used"
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 5,
        "color": "#630A10",
        "fontSize": 16,
        "height": 100,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    value="" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#F7D098",
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }>
    Water:
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Water Used"
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 5,
        "color": "#630A10",
        "height": 30,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    value="" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#F7D098",
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }>
    Post Brew Ingredients:
  </Text>
  <TextInput
    multiline={true}
    onChangeText={[Function]}
    placeholder="Add post brew ingredients"
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 5,
        "color": "#630A10",
        "fontSize": 16,
        "height": 100,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    value="" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#F7D098",
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }>
    Total Cost:
  </Text>
  <TextInput
    keyboardType="numbers-and-punctuation"
    onChangeText={[Function]}
    placeholder="Cost in Dollars"
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 5,
        "color": "#630A10",
        "height": 30,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    value="" />
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F7D098",
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }>
        Estimated Brew Date:
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderRadius": 5,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "padding": 5,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#630A10",
              "fontSize": 16,
            }
          }>
          November 9th 2016
        </Text>
      </View>
    </View>
    <View />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      (Click Date To Toggle Date Picker)
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F7D098",
          "borderRadius": 5,
          "height": 30,
          "justifyContent": "center",
          "margin": 20,
          "width": 300,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#630A10",
            "fontSize": 16,
          }
        }>
        Submit
      </Text>
    </View>
  </View>
</ScrollView>
`;
